### PR TITLE
Replaces toml with cosmiconfig

### LINKED ISSUE #248: Allow plugin directories to be optional
Using `gluegun@2.0.0-alpha.2`, I have this in my runtime config:

```ts
    .plugin(resolve(`${filesystem.cwd()}`, BRAND), LOAD_OPTIONS)
```

But I want that to be optional as it may not exist.

Here's the error it spits out:

```
Error: Error: couldn't load plugin (not a directory): /Users/steve/src/ir/app-machine/app-machine
    at loadFromDirectory (/Users/steve/src/ir/app-machine/node_modules/gluegun/src/loaders/toml-plugin-loader.js:40:11)
    at Runtime.load (/Users/steve/src/ir/app-machine/node_modules/gluegun/src/domain/runtime.js:184:20)
    at loadPlugins.forEach.entry (/Users/steve/src/ir/app-machine/node_modules/gluegun/src/domain/builder.js:56:19)
    at Array.forEach (<anonymous>)
    at Builder.create (/Users/steve/src/ir/app-machine/node_modules/gluegun/src/domain/builder.js:50:22)
    at Object.configureRuntime (/Users/steve/src/ir/app-machine/src/cli/gluegun-runtime.ts:20:6)
    at Object.<anonymous> (/Users/steve/src/ir/app-machine/src/cli/cli.ts:24:21)
    at step (/Users/steve/src/ir/app-machine/src/cli/cli.ts:32:23)
    at Object.next (/Users/steve/src/ir/app-machine/src/cli/cli.ts:13:53)
    at /Users/steve/src/ir/app-machine/src/cli/cli.ts:7:71
```

For `src` I can see why it's required, but for `plugin` and `plugins`, it should be optional.

### PR DESCRIPTION
**Pull request by jamonholmgren**
Additionally:
 * Doesn't throw if plugin() tries to load a missing folder
 * Removes Builder#on and #configFile options
 * Refactors some internal modules
 * Removes some unused fixtures

Fixes #248 and #282

### CODE REVIEW

File: docs/plugins.md [hunk 1]
```diff
@@ -10,11 +10,11 @@ A plugin is directory that contains 3 optional sub-directories:
 
 And 1 optional file:
 
-* `<brand>.toml`
+* `<brand>.config.js`
 
 Other than the 3 directories listed, you're welcome to put any other files or sub-directories in the plugin directory.
 
-Since multiple plugins can be loaded, they must have unique names.  The names are indicated by reading the `name` property of the `<brand>.toml` inside the plugin directory, or the plugin directory name itself.
+Since multiple plugins can be loaded, they must have unique names. The names are indicated by reading the `name` property of the object exported in `<brand>.config.js` inside the plugin directory, or the plugin directory name itself.
```
Comment (skellock) [inline docs/plugins.md]: Can you add the other config options here too?  .json and .yaml?

File: docs/runtime.md [hunk 2]
```diff
@@ -134,7 +128,7 @@ credits
     retrieve-imdb.js
   templates
     actor-view.js.ejs
-  gluegun.toml
+  credits.config.js
```
Comment (skellock) [inline docs/runtime.md]: Wouldn’t this be movie.config.js?

Comment (jamonholmgren) [reply docs/runtime.md]: I can change both, but the folder name is credits so I went with that.

**Pull request merged**